### ISSUE TITLE
about chamfer loss

### ISSUE THREAD

**Issue by qiuzidian** [aim-uofa/PM-Loss#3]
Hello, I would like to ask why the chamber loss uses single-directional instead of bi-directional?

**Comment by dc-walker** [comment]
Hi @qiuzidian ,

The reason we use a single-directional Chamfer loss is that we treat the pointmap as a "pseudo-ground truth" to guide the training. Our goal is to encourage the Gaussian centers predicted by the feed-forward 3DGS model to move towards the "better" geometric locations suggested by the pointmap prior. The loss is formulated to penalize the predicted Gaussian centers for deviating from this prior.

A bi-directional loss would also try to pull the clean pointmap prior towards the noisy 3DGS points. This is undesirable because it would corrupt our geometric guide with the very errors we are trying to fix.

**Comment by qiuzidian** [comment]
Thank you for your timely reply! However, as far as I know, single-directional Chamfer loss can cause the predicted point cloud to only partially cover the target, resulting in pattern collapse. "A bi-directional loss would also try to pull the clean pointmap prior to the noisy 3DGS points" I think this is inaccurate because VGGT has been frozen, and bi-directional loss will only encourage the prediction and target point cloud shape to be consistent. Have you conducted relevant ablation experiments?

**Comment by dc-walker** [comment]
Hi @qiuzidian, thanks for your insightful response.

The case in our work is quite specific, since we assume that the point clouds predicted by the feed-forward model $X_{3DGS}$ are *noisy*, while the pointmap-based ones $X_{PM}$ are *clean*. More importantly, both are obtained from a "pixel-aligned" prediction model using the same input images. Hence, we mainly aim to pull $X_{3DGS}$ closer to $X_{PM}$ ($X_{3DGS}$ -> $X_{PM}$), but we do not care if there is missing coverage ($X_{PM}$ -> $X_{3DGS}$), as this won't be the case since they are originally "pixel-aligned". This inherent alignment means we do not suffer from the typical partial coverage issue, as every point in $X_{3DGS}$ has a corresponding point in $X_{PM}$. My initial intention in mentioning "pull the clean pointmap prior towards the noisy 3DGS points" was mainly to highlight that we do not apply ($X_{PM}$ -> $X_{3DGS}$), we do not need to count this into the training loss as it is not necessary.

We will add an additional experiment to justify the choice of single direction over bi-direction in terms of quality and computational cost, and will keep you updated once we get the results. Thanks.

**Comment by qiuzidian** [comment]
Thanks for your clarification!
